math/src/unsigned_integer/element.rs

```diff
@@ -138,9 +138,8 @@ impl<const NUM_LIMBS: usize> Mul<&UnsignedInteger<NUM_LIMBS>> for &UnsignedInteg
                 t = NUM_LIMBS - 1 - i;
             }
         }
-
-        assert!(
-            n + t < NUM_LIMBS,
+        debug_assert!(
+            n + t + 1 < NUM_LIMBS,
```

Comment:
Maybe this extra `+ 1` is breaking it?
```suggestion
            n + t < NUM_LIMBS,
```

Reply:
Yes it was this extra +1, now it passed all the tests!